src/structures/BaseManagerFetchAll.ts

```diff
@@ -0,0 +1,29 @@
+import { FetchOptions, PaginationMeta } from "../common";
+import { BaseManager } from "./BaseManager";
+import { Dict } from "./Dict";
+
+export abstract class BaseManagerFetchAll<T extends unknown[], K, E> extends BaseManager {
+    abstract meta: PaginationMeta;
+
+    async fetchAll(...options: T): Promise<Dict<K, E>> {
```

Comment:
Is there a simpler way to do this? Rather than relying on parameter checks and the types...

Reply:
The only other way (that I can think of) to do this is to provide a protected method, which then gets called from inside each manager. Then it would be defined just like `fetch`, but it would call an internal method.
And of course, have an `fetchAll` abstract.

Reply:
Updated to the way I described above.